compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -1583,6 +1574,28 @@ impl<'a, 'tcx> Visitor<'tcx> for TypeChecker<'a, 'tcx> {
                                 },
                             )
                             .unwrap();
+                        } else if let ty::Dynamic(src_tty, src_lt) =
+                            *self.struct_tail(src.ty, location).kind()
+                            && let ty::Dynamic(dst_tty, dst_lt) =
+                                *self.struct_tail(dst.ty, location).kind()
+                            && src_tty.principal().is_none()
+                            && dst_tty.principal().is_none()
```

Comment:
@Darksonn I think this currently means that casting something like: `*mut dyn Trait + Send + 'a` to `*mut Send + 'b` won't check `'a: 'b`. We ought to drop the `src_tty.principal().is_none()` check here, unless I'm misreading this code :)